pennylane/transforms/decompose.py

```diff
@@ -858,21 +813,37 @@ def _operator_decomposition_gen(  # pylint: disable=too-many-arguments
     elif acceptance_function(op) or max_depth_reached:
         yield op
 
-    elif isinstance(op, (Allocate, Deallocate)):
-        yield op
-
-    elif graph_solution is not None and graph_solution.is_solved_for(op, num_available_work_wires):
+    elif graph_solution and graph_solution.is_solved_for(op, num_available_work_wires):
         op_rule = graph_solution.decomposition(op, num_available_work_wires)
         with queuing.AnnotatedQueue() as decomposed_ops:
             op_rule(*op.parameters, wires=op.wires, **op.hyperparameters)
         decomp = decomposed_ops.queue
-        current_depth += 1
         if num_available_work_wires is not None:
             num_available_work_wires -= op_rule.get_work_wire_spec(**op.resource_params).total
-    else:
+
+    elif enabled_graph() and isinstance(op, GlobalPhase):
+        warnings.warn(
+            "With qml.decomposition.enabled_graph(), GlobalPhase is not assumed to have a "
+            "decomposition. To disable this warning, add `GlobalPhase` to the gate set, or "
+            "assign a decomposition rule to `GlobalPhase` via the `fixed_decomps` keyword "
+            "argument. To make GlobalPhase decompose to nothing, you can import `null_decomp` "
+            "from pennylane.decomposition.decomposition_rule, and assign it to GlobalPhase."
```

Comment:
Oh interesting 🤔... didn't know this. @astralcai how do I use this?

Reply:
```python
from pennylane.decomposition.decomposition_rule import null_decomp

@partial(qml.transforms.decompose, gate_set={...}, fixed_decomps={qml.GlobalPhase: null_decomp})
@qml.qnode(...)
def circuit(...):
    ...
```

Reply:
Or even `{qml.GlobalPhase: lambda *args, **kwargs : return None}`?

Reply:
Right now a decomposition rule must have a resource function registered.

Reply:
Some things are back-ticked here and some things aren't 🤔 

Also, I can't find `decomposition.decomposition_rule` in the docs

Reply:
There is a decomposition rule implemented for a controlled global phase that prevents a controlled global phase to decompose to nothing even when global phase itself decomposes to nothing. @josh146